flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -980,8 +963,7 @@ protected CompletableFuture<Void> getCompletionFuture() {
 
     @Override
     public final void cancel() throws Exception {
-        isRunning = false;
-        canceled = true;
+        taskState.canceled = true;
```

Comment:
Shouldn't we also update `taskState.status` here?

(ideally, in my view we should use either enum or a set of flags inside the `TaskState`, but not both)

Reply:
> maintain the existing logic where both flags (running and canceled are true)

I see. Have you considered adding `CANCELLING` (in addition to `CANCELLED`) constant?